src/compute/src/server.rs

```diff
@@ -381,6 +381,12 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
                                 };
                             }
                             cap.downgrade(&(time + 1));
+                        } else {
+                            // Non-leader workers will still receive `UpdateConfiguration` commands
+                            // and we must drain those to not leak memory.
+                            if let Ok(_cmd) = command_rx.try_recv() {
+                                assert_ne!(worker_id, 0);
```

Comment:
I think we can enter this part only if
* We're not worker 0, because then we dropped our capability at the beginning,
* Or we're worker 0, received a disconnect, and then are scheduled again. This should never happen as it would mean the sender was dropped but someone held on to the activator.

For this reason, I don't think the assert adds much value, but a different assert would, see my other comment.

Reply:
Well, the same can be said about the existing assert on line 369. It might still be useful to have it in case we change the code around capability dropping but forget adjusting the code around the asserts. No strong feelings either way, except that we should have either both these asserts or none of them.

Reply:
Let's keep both. They should be very cheap.